zk/stages/stage_batches.go

```diff
@@ -256,7 +256,8 @@ LOOP:
 
 			atLeastOneBlockWritten = true
 
-			if l2Block.L2BlockNumber == 0 {
+			// ignore genesis or a repeat of the last block
+			if l2Block.L2BlockNumber == 0 || l2Block.L2BlockNumber == stageProgressBlockNo {
```

Comment:
What if we receive older blocks? Should we ignore them too? But maybe with a log warning?

Reply:
Yeah that's a fair shout.  I put this in to avoid the unwind, but it would be good to put in a warn for older blocks than expected

Reply:
Yeah, we probably need to split to two things like that:

```		
			if  l2Block.L2BlockNumber == 0 { continue }
			
			if l2Block.L2BlockNumber <= stageProgressBlockNo {
			    Log.Warn(“unexpected block numbers older than progress (received: L2BlockNumber, local: stageProgressBlockNo”)
			    continue
			 }
```

Reply:
We expect the latest block to arrive again, this was part of the changes to avoid the unwind, so I have put the warning only for anything older than that

Reply:
Ah, okay, sure

Reply:
Just make sure you comment the code then, it is not super obvious tbh